packages/core/src/gitlab.ts

```diff
@@ -0,0 +1,204 @@
+/* eslint-disable camelcase, @typescript-eslint/camelcase */
+
+import * as GitlabApi from 'gitlab';
+import * as debug from 'debug';
+import * as envCi from 'env-ci';
+import { buildKeyedComment } from './helpers';
+
+interface CommentlyArgs {
+  /** The MR sha to comment on. Is detected in CI environments */
+  commit?: string;
+  /** The title at the top of the comment */
+  title?: string;
+  /** The unique key to identify the comment by, not shown to end users */
+  key?: string;
+  /** Key a history of the comments in the comment created by this library */
+  useHistory?: boolean;
+  /** The gitlab token if not in your environment */
+  token?: string;
+}
+
+interface User {
+  id: number;
+}
+
+interface MRInfo {
+  project_id: number;
+  iid: number;
+  sha: string;
+}
+
+export interface Comment {
+  author: {
+    id: number;
+  };
+  body: string;
+  sha: string;
+  id: number;
+}
+
+/** Create a "commenter" that can comment on a pull request */
+export class Gitlab {
```

Comment:
Does it make sense for `GitLab` and `GitHub` to extend from the same interface so we don't redeclare these internal bits?